### PR TITLE
docs: ec2: Add documentation to the `vpcEndpointDnsEntries` prop

### PR DESCRIPTION
**Pull request by gshpychka**
I mistakenly assumed this prop would contain the list of DNS names and did not realize it is actually a list of the hosted zone ID and the DNS name separated by `:`. This change includes a part of the CloudFormation documentation to the attribute to help others avoid this mistake.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts [hunk 1]
```diff
@@ -520,6 +520,9 @@ export class InterfaceVpcEndpoint extends VpcEndpoint implements IInterfaceVpcEn
 
   /**
    * The DNS entries for the interface VPC endpoint.
+   * Each entry is a combination of the hosted zone ID and the DNS name. 
+   * The entries are ordered as follows: regional public DNS, zonal public DNS, private DNS, and wildcard DNS. 
+   * This order is not enforced for AWS Marketplace services.
```
Comment (corymhall) [inline packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts:525]:
```suggestion
   * This order is not enforced for AWS Marketplace services.
   *
   * The following is an example. In the first entry, the hosted zone ID is Z1HUB23UULQXV 
   * and the DNS name is vpce-01abc23456de78f9g-12abccd3.ec2.us-east-1.vpce.amazonaws.com.
   *
   * ["Z1HUB23UULQXV:vpce-01abc23456de78f9g-12abccd3.ec2.us-east-1.vpce.amazonaws.com",
   * "Z1HUB23UULQXV:vpce-01abc23456de78f9g-12abccd3-us-east-1a.ec2.us-east-1.vpce.amazonaws.com",
   * "Z1C12344VYDITB0:ec2.us-east-1.amazonaws.com"]
   *
   * If you update the PrivateDnsEnabled or SubnetIds properties, the DNS entries in the list will change.
```

**Review by corymhall — CHANGES_REQUESTED** [review]
Very helpful! Just one suggestion to add the rest of the CFN docs since I think the examples are beneficial.

File: packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts:525]: Don't forget to backquote these code samples for Markdown

Comment (gshpychka) [reply packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts:525]: @rix0rrr can you please clarify what you mean? You mean I should surround the sample with ```...```?

**Review by corymhall — APPROVED** [review]

**Pull request merged**